vavr/src/test/java/io/vavr/MatchTest.java

```diff
@@ -318,9 +320,9 @@ public void shouldDecomposeListOfTuple3() {
     @SuppressWarnings("UnnecessaryLocalVariable")
     @Test
     public void shouldDecomposeListWithNonEmptyTail() {
-        final List<Option<Number>> intOptionList = List.of(Option.some(1), Option.some(2.0));
-        final String actual = Match(intOptionList).of(
-                Case($Cons($Some($(1)), $Cons($Some($(2.0)), $())), (x, xs) -> {
+        final List<Option<Number>> numberOptionList = List.of(Option.some(1), Option.some(2.0));
+        final String actual = Match(numberOptionList).of(
+                Case($Cons($Some($(1)), $Cons($Some($(2.0)), $())), (Some<Number> x, List<Option<Number>> xs) -> {
```

Comment:
@danieldietrich, it only compiles on `Java 9` if we explicitly specify the types here. Is that still ok?